### ISSUE TITLE
0xpranav - SuperPool.sol#togglePause() function doesn't pause/unpause superpool functionality

### ISSUE THREAD

**Issue by sherlock-admin2** [sherlock-audit/2024-08-sentiment-v2-judging#83]
0xpranav

Medium

# SuperPool.sol#togglePause() function doesn't pause/unpause superpool functionality

### Summary

In superpool contract, it inherits Pausable contract and the togglePause function is used to pause/unpause the state of the superpool but it does nothing because whenNotPause() and whenPause() modifiers are not used anywhere.

### Root Cause

 According to the openzeppelin Pausable contract,  
*This module is used through inheritance. It will make available the modifiers `whenNotPaused` and `whenPaused`, which can be applied to the functions of your contract. Note that they will not be pausable by simply including this module, only once the modifiers are put in place.* [See here](https://docs.openzeppelin.com/contracts/5.x/api/utils#Pausable)
In SuperPool.sol#togglePause() , it calls pause() and unpause() to pause/unpause the state of the SuperPool(https://github.com/sherlock-audit/2024-08-sentiment-v2/blob/main/protocol-v2/src/SuperPool.sol#L163). But there are no implementation of whenNotPaused and whenPaused modifers in any functions. Thus owner cannot the pause / unpause the superPool 

### Impact

Owner cannot pause or unpause the superpool.

### Mitigation

To use the functionality of Pausable contract simply add whenNotPaused or whenPaused modifiers in necessary functions 

Duplicate of #270
